test: add coverage for `node` field of `getaddednodeinfo` RPC

brunoerg opened the pull request:
We currently do not test a successful call to `getaddednodeinfo` filtering by `node`, we only test it with an unknown address and checks whether it fails. This PR adds coverage to it.

brunoerg committed:
test: add coverage for `node` field of `getaddednodeinfo` RPC

tdb3 commented on test/functional/rpc_net.py:
```diff
@@ -249,11 +249,17 @@ def test_addnode_getaddednodeinfo(self):
         assert_equal(added_nodes[0]['addednode'], ip_port)
+        # check that filtering by node works
+        self.nodes[0].addnode(node="11.22.33.44", command='add')
+        added_node = self.nodes[0].getaddednodeinfo(node=ip_port)
```
nit:  If this file is touched, would recommend using `first_added_node` or `added_node_0` instead of `added_node`  to make it clearer that the node being referred to is the one added before 11.22.33.44.  Alternatively, could update the comment/log message instead.

brunoerg replied:
`added_node_0` seems weird because it can be confused with nodes[0], I prefer `first_added_node`. I will address it.

tdb3 commented on test/functional/rpc_net.py:
```diff
@@ -249,11 +249,17 @@ def test_addnode_getaddednodeinfo(self):
         added_nodes = self.nodes[0].getaddednodeinfo()
         assert_equal(len(added_nodes), 1)
         assert_equal(added_nodes[0]['addednode'], ip_port)
+        # check that filtering by node works
```
nit: How about making this comment a log message instead?

https://github.com/bitcoin/bitcoin/blob/master/test/functional/README.md#general-test-writing-advice

> Instead of inline comments or no test documentation at all, log the comments to the test log, e.g. self.log.info('Create enough transactions to fill a block'). Logs make the test code easier to read and the test logic easier [to debug](https://github.com/bitcoin/bitcoin/blob/master/test/README.md#test-logging).

Also, since we're touching this function, it might be reasonable to add a commit to convert the surrounding comments of this function to log messages.  Any objections from others?

brunoerg replied:
It seems good to me, I'll address it.

brunoerg replied:
Done.

tdb3 commented on test/functional/rpc_net.py:
```diff
@@ -249,11 +249,17 @@ def test_addnode_getaddednodeinfo(self):
+        # check that filtering by node works
+        self.nodes[0].addnode(node="11.22.33.44", command='add')
+        added_node = self.nodes[0].getaddednodeinfo(node=ip_port)
+        assert_equal(added_nodes, added_node)
```
Manually induced a failure here by using `getaddednodeinfo(node="192.168.99.99")`.  Test failed as expected.

tdb3 reviewed (commented):
Approach ACK

Good update to exercise the filtering feature of the RPC.
Left a suggestion and nit.

kevkevinpal commented on test/functional/rpc_net.py:
```diff
@@ -249,11 +249,17 @@ def test_addnode_getaddednodeinfo(self):
         # check that node can be removed
         self.nodes[0].addnode(node=ip_port, command='remove')
-        assert_equal(self.nodes[0].getaddednodeinfo(), [])
+        added_nodes = self.nodes[0].getaddednodeinfo()
+        assert_equal(len(added_nodes), 1)
+        assert_equal(added_nodes[0]['addednode'], "11.22.33.44")
```
instead of modifying the code here why not use `self.nodes[0].addnode(node="11.22.33.44", command='remove')` instead?

brunoerg replied:
Because I wanted to remove exactly the first added node.

kevkevinpal commented on test/functional/rpc_net.py:
```diff
@@ -249,11 +249,17 @@ def test_addnode_getaddednodeinfo(self):
+        # check that filtering by node works
+        self.nodes[0].addnode(node="11.22.33.44", command='add')
+        added_node = self.nodes[0].getaddednodeinfo(node=ip_port)
+        assert_equal(added_nodes, added_node)
```
I'm not sure if we test for any list of added_nodes greater than a length of 1

brunoerg replied:
Yes, I can add this verification.

brunoerg replied:
Done.

kevkevinpal reviewed (commented):
Concept ACK [13a871f](https://github.com/bitcoin/bitcoin/pull/30339/commits/13a871f8e3835355b84ba8a404870dcb33441aef)

ran locally and it passed for me

brunoerg committed:
test: change comments to `self.log.info` for `test_addnode_getaddednodeinfo`

brunoerg commented:
Force-pushed:

- I added a commit to change some comments to `self.log.info`
- Addressed: https://github.com/bitcoin/bitcoin/pull/30339#discussion_r1653913528 and https://github.com/bitcoin/bitcoin/pull/30339#discussion_r1653781334

Thanks, @kevkevinpal and @tdb3.

kevkevinpal commented:
ACK [e38eadb](https://github.com/bitcoin/bitcoin/pull/30339/commits/e38eadb2c2d93d2ee3c9accb649b2de144b3732e)[e38eadb](https://github.com/bitcoin/bitcoin/pull/30339/commits/e38eadb2c2d93d2ee3c9accb649b2de144b3732e)

tdb3 reviewed (approved):
re ACK e38eadb2c2d93d2ee3c9accb649b2de144b3732e
Thanks for implementing the changes.

BrandonOdiwuor reviewed (approved):
Code Review ACK e38eadb2c2d93d2ee3c9accb649b2de144b3732e

rkrux reviewed (approved):
tACK [e38eadb](https://github.com/bitcoin/bitcoin/pull/30339/commits/e38eadb2c2d93d2ee3c9accb649b2de144b3732e)

`make, test/functional` are successful.
Thanks for testing the filtering by node functionality, and replacing the comments with  info/debug logs as suggested by @tdb3.

achow101 commented:
ACK e38eadb2c2d93d2ee3c9accb649b2de144b3732e
